src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -312,58 +315,89 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
+    gates.connect_cell_pair((1, 6), (13, 0)); // product1_hi_0
+                                              // Witness updated below
+
+    // Add witness for external multi-range checks:
+    // [quotient0, quotient1, quotient2]
+    // [quotient_bound, product1_lo, product1_hi_0]
+    external_checks.extend_witness_multi_range_checks(&mut witness);
+
+    // DESIGNER CHOICE: left and right (and result bound from before)
+    let left_limbs = left_input.to_field_limbs();
+    let right_limbs = right_input.to_field_limbs();
+    // Constant Double Generic gate for result and quotient bounds
+    external_checks.add_high_bound_computation(&left_limbs[2]);
+    external_checks.add_high_bound_computation(&right_limbs[2]);
+    CircuitGate::extend_high_bounds(&mut gates, &mut next_row, &foreign_field_modulus);
+    gates.connect_cell_pair((15, 0), (0, 2)); // left2
+    gates.connect_cell_pair((15, 3), (0, 5)); // right2
+    external_checks.extend_witness_high_bounds_computation(&mut witness, &foreign_field_modulus);
 
-    // Multiplication witness value product1_lo, product1_hi_0, carry1_lo multi-range-check
+    // Left input multi-range-check
+    external_checks.add_multi_range_check(&left_limbs);
     CircuitGate::extend_multi_range_check(&mut gates, &mut next_row);
-    gates.connect_cell_pair((0, 6), (12, 0)); // carry1_lo
-    gates.connect_cell_pair((1, 5), (13, 0)); // product1_lo
-    gates.connect_cell_pair((1, 6), (14, 0)); // product1_hi_0
+    gates.connect_cell_pair((0, 0), (16, 0)); // left_input0
+    gates.connect_cell_pair((0, 1), (17, 0)); // left_input1
+    gates.connect_cell_pair((0, 2), (18, 0)); // left_input2
                                               // Witness updated below
 
-    // Result/remainder bound multi-range-check
+    // Right input multi-range-check
+    external_checks.add_multi_range_check(&right_limbs);
     CircuitGate::extend_multi_range_check(&mut gates, &mut next_row);
-    gates.connect_cell_pair((3, 0), (16, 0));
-    gates.connect_cell_pair((3, 1), (17, 0));
-    gates.connect_cell_pair((3, 2), (18, 0));
-    // Witness updated below
+    gates.connect_cell_pair((0, 3), (20, 0)); // right_input0
+    gates.connect_cell_pair((0, 4), (21, 0)); // right_input1
+    gates.connect_cell_pair((0, 5), (22, 0)); // right_input2
+                                              // Witness updated below
 
-    // Add witness for external multi-range checks (product1_lo, product1_hi_0, carry1_lo and result)
+    // Add witness for external multi-range checks:
+    // left and right limbs
     external_checks.extend_witness_multi_range_checks(&mut witness);
 
-    // Quotient bound multi-range-check
-    CircuitGate::extend_compact_multi_range_check(&mut gates, &mut next_row);
-    gates.connect_cell_pair((1, 3), (22, 1));
-    gates.connect_cell_pair((1, 4), (20, 0));
-    external_checks.extend_witness_compact_multi_range_checks(&mut witness);
+    // [result_bound, 0, 0]
```

Comment:
Why not put `left2` and `right2` in here so it matches the comment above with the Layout?  Either that our update the layout comment.